core/src/impl/Kokkos_ViewMapping.hpp

```diff
@@ -2530,21 +2529,12 @@ struct ViewDataHandle<
 namespace Kokkos {
 namespace Impl {
 
+inline static constexpr std::byte all_zeroes[256] = {};
+
 template <typename T>
-inline bool is_zero_byte(const T& t) {
-  using comparison_type = std::conditional_t<
-      sizeof(T) % sizeof(long long int) == 0, long long int,
-      std::conditional_t<
-          sizeof(T) % sizeof(long int) == 0, long int,
-          std::conditional_t<
-              sizeof(T) % sizeof(int) == 0, int,
-              std::conditional_t<sizeof(T) % sizeof(short int) == 0, short int,
-                                 char>>>>;
-  auto bit_values = Kokkos::bit_cast<
-      Kokkos::Array<comparison_type, sizeof(T) / sizeof(comparison_type)>>(t);
-  for (std::size_t i = 0; i < sizeof(T) / sizeof(comparison_type); ++i)
-    if (bit_values[i] != 0) return false;
-  return true;
+bool is_zero_byte(const T& x) {
+  if (sizeof(T) > sizeof(all_zeroes)) return false;  // out of luck
```

Comment:
I'd prefer
```suggestion
  constexpr std::byte all_zeroes[sizeof(T)] = {};
```
to be more general. The generated instructions for all versions are pretty similar, see https://godbolt.org/z/6Ef97eKhv.